File: RedBusApp/src/app/models/class/Bus.ts
export class Bus{
    id: string;
    name: string;
    fromLocationId: string;
    toLocationId: string;
    departTime: string;
    arriveTime: string;
    totalCapacity: number;
    price: number;
    constructor(){
        this.id = "";
        this.name = "";
        this.fromLocationId = "";
        this.toLocationId = "";
        this.departTime = "";
        this.arriveTime = "";
        this.totalCapacity = 0;
        this.price = 0;
    }
}

File: APIProject/APIProject/Models/Bus.cs
using System.ComponentModel.DataAnnotations;

namespace APIProject.Models
{
    public class Bus
    {
        public string Id { get; set; } = Guid.NewGuid().ToString();
        [Required]
        public string Name { get; set; }
        [Required]
        public string FromLocationId { get; set; }
        [Required]
        public string ToLocationId { get; set; }
        [Required]
        public DateTime DepartTime { get; set; }
        [Required]
        public DateTime ArriveTime { get; set; }
        [Required]
        public int TotalCapacity { get; set; }
        [Required]
        public int Price { get; set; }
    }
}